src/razor/src/completion/razorCompletionItemProvider.ts

```diff
@@ -278,6 +287,38 @@ export class RazorCompletionItemProvider extends RazorLanguageFeatureBase implem
 
         return item;
     }
+
+    private static addUsingKeyword(
+        language: LanguageKind,
+        razorDocument: vscode.TextDocument,
+        hostDocumentPosition: vscode.Position,
+        completionItems: vscode.CompletionItem[]
+    ) {
+        // This is an ugly hack, but it's needed to get the "using" keyword to show up in the completion list.
+        // The reason it doesn't show up is because the C# generated document puts the position of the cursor
+        // at '__o = [||]', which isn't a valid location for a using statement.
+        if (language == LanguageKind.CSharp) {
+            const line = razorDocument.lineAt(hostDocumentPosition.line);
+            const lineText = line.text.substring(0, hostDocumentPosition.character);
+            if (
+                lineText.endsWith('@') ||
```

Comment:
I know this check looks terrible - honestly it makes me cringe too 😅 If anyone has suggestions for an alternative, feel free to suggest

Reply:
This works but... ugh... 

What's the benefit of having a using snippet? Can we just make a server fix for Razor to add it's own completion for `@` directives? I think we already do have some... (like `@co`)

Reply:
And https://github.com/dotnet/razor/pull/6012/files#diff-052d5834149367e2bcf4d95d0740ddb4b346116cfcd64e568d0b5eca38f7b8aeL739

Reply:
> What's the benefit of having a using snippet? Can we just make a server fix for Razor to add it's own completion for @ directives? I think we already do have some... 

I'm guessing a using snippet might be beneficial in the case where the user is already in a C# context and would want to insert something like `using () { }`?

I've filed https://github.com/dotnet/razor/issues/9330 to track a (still hacky) server-side fix that we can use in both VS and VS Code. In the interest of getting a temporary fix in I'm planning on merging this version for now, but will follow up soon on the server-side fix.